File: MyDictionary.Api/Contracts/StudyDecks/Responses/StudyDeckDetailResponse.cs
namespace MyDictionary.Api.Contracts.StudyDecks.Responses;

public record StudyDeckDictionaryResponse(
    Guid DictionaryId,
    string Name,
    bool IsSynchronized
);

public record StudyDeckDetailResponse(
    Guid Id,
    string Name,
    string? Description,
    List<StudyDeckDictionaryResponse> Dictionaries
);

File: MyDictionary.Api/ClientApp/src/features/StudyDeck/types/responses/StudyDeckDetailResponse.ts
export interface StudyDeckDictionaryResponse {
    dictionaryId: string;
    name: string;
    isSynchronized: boolean;
}

export interface StudyDeckDetailResponse {
    id: string;
    name: string;
    description?: string;
    dictionaries: StudyDeckDictionaryResponse[];
}
